GoogleSignIn/Tests/Unit/GIDAuthorizationResponseHelperTest.m

```diff
@@ -92,78 +101,89 @@ - (void)testGenerateAuthFlowFromAuthorizationResponse_noCode {
                                                                   emmSupport:nil
-                                                                    flowName:GIDFlowNameVerify
+                                                                    flowName:GIDFlowNameVerifyAccountDetail
                                                                configuration:nil
                                                                        error:nil];
 
   GIDAuthFlow *authFlow = [responseHandler generateAuthFlowFromAuthorizationResponse];
   XCTAssertNotNil(authFlow);
   XCTAssertNil(authFlow.authState);
   XCTAssertNotNil(authFlow.error);
+  XCTAssertEqual(authFlow.error.domain, expectedError.domain);
   XCTAssertEqual(authFlow.error.code, expectedError.code);
-  XCTAssertEqualObjects(authFlow.error.userInfo[NSLocalizedDescriptionKey],
-                        expectedError.userInfo[NSLocalizedDescriptionKey]);
-#endif // TARGET_OS_IOS && !TARGET_OS_MACCATALYST
+  XCTAssertEqualObjects(authFlow.error.userInfo, expectedError.userInfo);
 }
+#endif // TARGET_OS_IOS && !TARGET_OS_MACCATALYST
 
-- (void)testGenerateAuthFlowWithMissingAuthorizationResponse {
 #if TARGET_OS_IOS && !TARGET_OS_MACCATALYST
+- (void)testGenerateAuthFlowWithMissingAuthorizationResponse {
   NSError *error = [NSError errorWithDomain:kGIDVerifyErrorDomain
                                        code:GIDVerifyErrorCodeUnknown
                                    userInfo:nil];
+
+  NSString *errorString = [error localizedDescription];
+  NSDictionary<NSString *, NSString *> *errorDict = @{ NSLocalizedDescriptionKey : errorString };
+  NSError *expectedError = [NSError errorWithDomain:kGIDVerifyErrorDomain
+                                               code:GIDVerifyErrorCodeUnknown
+                                           userInfo:errorDict];
+
   GIDAuthorizationResponseHandler *responseHandler =
       [[GIDAuthorizationResponseHandler alloc] initWithAuthorizationResponse:nil
                                                                   emmSupport:nil
-                                                                    flowName:GIDFlowNameVerify
+                                                                    flowName:GIDFlowNameVerifyAccountDetail
                                                                configuration:nil
                                                                        error:error];
 
   GIDAuthFlow *authFlow = [responseHandler generateAuthFlowFromAuthorizationResponse];
   XCTAssertNotNil(authFlow);
   XCTAssertNil(authFlow.authState);
   XCTAssertNotNil(authFlow.error);
-  XCTAssertEqual(authFlow.error.code, error.code);
-#endif // TARGET_OS_IOS && !TARGET_OS_MACCATALYST
+  XCTAssertEqual(authFlow.error.domain, expectedError.domain);
+  XCTAssertEqual(authFlow.error.code, expectedError.code);
+  XCTAssertEqualObjects(authFlow.error.userInfo, expectedError.userInfo);
```

Comment:
Are you able to simplify these to just `XCTAssertEqual(authFlow.error, expectedError)`?

Reply:
Unfortunately doing this results in an error even with the values being the same. I could create an `isEqual:` if preferred.